.github/CODEOWNERS

```diff
@@ -1,5 +1,30 @@
-* @DataDog/apm-ruby
+* @DataDog/ruby-guild
 
 # Public Documentation
 /docs/Compatibility.md @DataDog/documentation
-/docs/GettingStarted.md @DataDog/documentation
\ No newline at end of file
+/docs/GettingStarted.md @DataDog/documentation
+
+# Library
+lib/datadog/appsec/* @DataDog/asm-ruby
+lib/datadog/appsec.rb @DataDog/asm-ruby
+lib/datadog/tracing/* @DataDog/apm-ruby
+lib/datadog/tracing.rb @DataDog/apm-ruby
+lib/datadog/opentelemetry/* @DataDog/apm-ruby
+lib/datadog/opentelemetry.rb @DataDog/apm-ruby
+lib/datadog/opentracer/* @DataDog/apm-ruby
+lib/datadog/opentracer.rb @DataDog/apm-ruby
+lib-injection/* @DataDog/apm-ruby
+lib/datadog/profiling/* @DataDog/profiling-rb
+lib/datadog/profiling.rb @DataDog/profiling-rb
+ext/* @DataDog/profiling-rb
```

Comment:
This question is probably outside the scope of this specific PR, but should we rename `ext/` since it appears to be a profiling only folder?

Reply:
We can't as `ext/` is a standard for Ruby gems with native extensions: https://guides.rubygems.org/gems-with-extensions/

Reply:
My rationale here is:

- `ext` being a generic folder for extensions, it should be owned by the guild.
- `ext/ddtrace_profiling_loader` and `ext/ddtrace_profiling_native_extension` should be owned by `profiling-rb`

Reply:
That said, currently profiling is the sole user of `libdatadog`, but that may/should change, therefore I would like to encourage ownership to be split:

- extension code pertaining to profiling should live in its own profiling-owned subdirectories
- extension code pertaining to libdatadog in general should live in guild-owned subdirectories